ckan/lib/search/index.py

```diff
@@ -236,6 +236,9 @@ def index_package(self,
                 except DateParserError:
                     log.warning('%r: %r value of %r is not a valid date', pkg_dict['id'], key, value)
                     continue
+            if isinstance(value, dict):
+                # Solr 8+ chokes on passing dict values unless doing an atomic update
```

Comment:
we are on solr9, is this still needed?

Reply:
I think we have updated the affected plugins so they don't pass JSON anymore, which would make this trickier to test, but it's for future-proofing. And it's very likely that Solr 9 is still affected; this wasn't considered a bug.